### ISSUE TITLE
Duplicate file check not right

### ISSUE THREAD

**Issue by weidercs** [evilhero/mylar#1575]
I have an issue where a few issues each pull aren't being caught by the duplicate file check.  I copy the files so that I can continue to seed them...  Is there a setting I should change or is it a bug?

![mylar_-_history](https://cloud.githubusercontent.com/assets/1444151/23226124/53c74246-f8e9-11e6-877e-d2fad15ddd53.jpg)

```
22-Feb-2017 10:22:05 - INFO    :: FOLDERMONITOR : [DUPECHECK] Duplicate check for /var/nfs/torrent/complete/comic/Infamous Iron Man 005 (2017) (digital) (Minutemen-Thoth).cbz
22-Feb-2017 10:22:05 - INFO    :: FOLDERMONITOR : [DUPECHECK] Existing Status already set to Downloaded
22-Feb-2017 10:22:05 - INFO    :: FOLDERMONITOR : [DUPECHECK] Existing file within db :Infamous Iron Man 005 (2017).cbz has a filesize of : 41136263 bytes.
22-Feb-2017 10:22:05 - DEBUG   :: FOLDERMONITOR : [DUPECHECK] Based on duplication preferences I will retain based on : filesize
22-Feb-2017 10:22:05 - INFO    :: FOLDERMONITOR : [DUPECHECK-FILESIZE PRIORITY] [#5] Retaining newly scanned in filename : /var/nfs/torrent/complete/comic/Infamous Iron Man 005 (2017) (digital) (Minutemen-Thoth).cbz
22-Feb-2017 10:22:05 - INFO    :: FOLDERMONITOR : [POST-PROCESSING] [9/22] Starting Post-Processing for Infamous Iron Man issue: 5
22-Feb-2017 10:22:05 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] issueid: 582533
22-Feb-2017 10:22:05 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] issuenumOG: 5
22-Feb-2017 10:22:05 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] Zero Suppression set to : 00x
22-Feb-2017 10:22:05 - DEBUG   :: FOLDERMONITOR : issue detected less than 10
22-Feb-2017 10:22:05 - DEBUG   :: FOLDERMONITOR : Zero level supplement set to 00x. Issue will be set as : 005
22-Feb-2017 10:22:05 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] Pretty Comic Issue is : 005
22-Feb-2017 10:22:05 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] Issue Year : 2017
22-Feb-2017 10:22:05 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] Publisher: Marvel
22-Feb-2017 10:22:05 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] Series: Infamous Iron Man
22-Feb-2017 10:22:05 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] Year: 2016
22-Feb-2017 10:22:05 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] Comic Location: /mnt/Media/Comics/Infamous Iron Man (2016)
22-Feb-2017 10:22:05 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] Comic Version: None
22-Feb-2017 10:22:05 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] No version # found for series, removing from filename
22-Feb-2017 10:22:05 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] New format is now: $Series $Issue ($Year)
22-Feb-2017 10:22:05 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] Not an annual - removing from filename parameters
22-Feb-2017 10:22:05 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] New format: $Series $Issue ($Year)
22-Feb-2017 10:22:05 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] Metatagging enabled - proceeding...
```

**Comment by evilhero** [comment]
It's probably because the copy option is retaining the file in the scanned folder and keeps on trying to post-process it over & over. I'll have to try and add some kind of check for an additional option of some kind if the copy option is enabled.

**Comment by weidercs** [comment]
Thanks, it worked previously, I just started noticing this a few weeks ago but have been copying the files since I can remember.  And of the 15 or so issues it downloaded it's only redoing the post processing for 4 over and over.

**Comment by weidercs** [comment]
I dug into this a little more, it's doing the filesize check and for some reason the file post processing is actually smaller than the original (both cbz).  I'm assuming the system does a check to see if the filesize is bigger and if it is, tires to process it again.  Odd that the filesize would get smaller with processing, but that's what is happening.

**Comment by weidercs** [comment]
original file:
```
49208778 Mar  8 11:48 Jessica Jones 006 (2017) (digital) (Minutemen-Faessla).cbz
```
processed file:
```
49198796 Mar  8 13:16 /mnt/Media/Comics/Jessica.Jones.(2016)/Jessica Jones 006 (2017).cbz
```

**Comment by evilhero** [comment]
Yeah that's weird - I mean if the 2 files exist on different partitions (like if one parition was ext4 and another was ntfs, kinda thing), that would be the only reason I can think of for the small discrepency in size. Or another possibility, the metadata has changed since tagging the original and the newest one contains the newest metadata which is causing some bytes difference in the filesizes (because it saves it as zip comment, and a comicinfo.xml file within the cbz if you have those options enabled).

I haven't forgotten about this, just things have been crazy hectic here so I've had to minimize my time with Mylar - but think things should be looking good now. 

I'm assuming your intention is to keep the original file, and just ignore the post-processed one (as in don't post-process it at all, and retain it for seeding). This gets kinda tricky with metatagging - if a file is seeding and a cbr (originally),  it would have to know that the file that's already been post-processed (and is now a cbz with metatagging) is actually the cbr converted. If it could link the 2, then it would know not to post-process and could ignore it. Even if it's a cbz originally, the metadata could change within the file which could account for some bytes difference. 

If the last point is probably what we're trying to do (and I think we are), I'll have to figure out a way to create a link between the 2 to avoid post-processing the same issue repeatedly.

**Comment by weidercs** [comment]
no rush, just noticed it on the latest pull again so I dug into it...  Yes, intention is to keep seeding the file, I keep the ratio pretty high so I can give back to the community.

One solution is you 'could' save the file size to the db of the unprocessed file and then check when you pick it up that it's the same file (or hash if you want to get fancy, but might add some overhead to hash the file).  Then ignore it if it's already been successfully processed.  I'm sure there's some scenarios that I'm not thinking of that it wouldn't work for but planting a seed.

Thanks for all the hard work, your platform continues to be amazing.

**Comment by weidercs** [comment]
Fixed with [4d5b384](https://github.com/evilhero/mylar/commit/4d5b384c825902ef6289d4ac91be1a6a6867552f)
